tests/php/Authenticator/ChangePasswordHandlerTest.php

```diff
@@ -0,0 +1,110 @@
+
+        Injector::inst()->load([
+            Security::class => [
+                'properties' => [
+                    'authenticators' => [
+                        'default' => '%$' . MemberAuthenticator::class,
+                    ]
+                ]
+            ],
+            LoggerInterface::class . '.mfa' => [
+                'class' => 'Monolog\Handler\NullHandler'
+            ],
+        ]);
+    }
+
+    /**
+     * @param Member $member
+     * @param string $password
+     * @return HTTPResponse
+     */
+    protected function doLogin(Member $member, $password)
+    {
+        $this->get('Security/changepassword');
+
+        return $this->submitForm(
+            'MemberLoginForm_LoginForm',
+            null,
+            array(
+                'Email' => $member->Email,
+                'Password' => $password,
+                'AuthenticationMethod' => MemberAuthenticator::class,
+                'action_doLogin' => 1,
+            )
+        );
+    }
+
+    public function testMFADoesNotLoadWhenAUserIsLoggedIn()
+    {
+        /** @var Member&MemberExtension $member */
+        $member = $this->objFromFixture(Member::class, 'simon');
+        $this->logInAs($member);
+        $this->get('Security/changepassword');
+        $this->assertNotEmpty($this->cssParser()->getByXpath('//input[@type="password"][@name="OldPassword"]'));
+    }
+
+    public function testMFADoesNotLoadWhenAUserDoesNotHaveRegisteredMethods()
+    {
+        /** @var Member&MemberExtension $member */
+        $member = $this->objFromFixture(Member::class, 'guy');
+        $m = $member->ID;
+        $t = $member->generateAutologinTokenAndStoreHash();
+        $this->get("Security/changepassword?m={$m}&t={$t}");
+        $parser = $this->cssParser();
+        $this->assertNotEmpty(
+            $parser->getByXpath('//input[@type="password"][@name="NewPassword1"]'),
+            'There should be a new password field'
+        );
+        $this->assertNotEmpty(
+            $parser->getByXpath('//input[@type="password"][@name="NewPassword2"]'),
+            'There should be a confirm new password field'
+        );
+    }
+
+    public function testMFALoadsWhenAUserHasConfiguredMethods()
+    {
+        /** @var Member&MemberExtension $member */
+        $member = $this->objFromFixture(Member::class, 'robbie');
+        $m = $member->ID;
+        $t = $member->generateAutologinTokenAndStoreHash();
+        $this->get("Security/changepassword?m={$m}&t={$t}");
+        $parser = $this->cssParser();
+        $this->assertEmpty($parser->getByXpath('//input[@type="password"]'));
+        $mfaApp = $parser->getBySelector('#mfa-app');
+        $this->assertNotEmpty($mfaApp);
+        $this->assertCount(1, $mfaApp);
+        $this->assertArraySubset(
+            ['data-schemaurl' => "/Security/changepassword/mfa/schema"],
+            current($mfaApp[0]->attributes())
+        );
```

Comment:
I think a lot of these assertions could be replaced with `assert[Not]Contains('Some text you're expecting in the response', $response->getBody())` to be honest